### PR TITLE
Feedback of unexpected/error cases.

### PR DESCRIPTION
**Pull request by vogdb**
For example.
-------------------------------------------------
```json
{
  "janus" : "error",
    "transaction" : "--------"
    "error" : {
      "code" : 405
      "reason" : "Illegal plugin to access"
    }
}
```
- ProxyConnection.onAttach:

  ```js
  if (!this.isAllowedPlugin(message.plugin)) {
    return Promise.reject('Disallowed plugin ' + message.plugin);
  }
  ```
- ProxyConnection.pluginFactory:

  ```js
    if (!pluginClass) {
      throw new Error('Invalid plugin');
    }
  ```

----------------------------------------------------------------------------------
```json
{
  "janus" : "error",
    "transaction" : "--------"
    "error" : {
      "code" : 403
      "reason" : "Unauthorized request"
    }
}
```

- BrowserConnection._onMessage:

  ```js
  auth.authorizeConnection(this, token)
  ```
- BrowserConnection._send:

  ```js
  auth.isValidConnection(this)
  ```

- PluginStreaming.onCreate:

  ```js
  auth.canPublish
  ```

- PluginStreaming.onWatch:

  ```js
  auth.canSubscribe
  ```

--------------------------------------------------------------

- ProxyConnection.onCreate. Just remove the error case.
- ProxyConnection.onDestroy. Remove it. Clear sessionId in hangup.

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review.
Some thoughts to discuss.
- Should we also pass an argument `message` to `JanusError`?
- Should we unify the error messages for `logger.error` and `JanusError`?
- I didn't transform `new Error('Invalid plugin')` at `ProxyConnection.prototype.pluginFactory` cause it does not relate to janus IMO. Wdyt about it?

@fvovan FYI

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/proxy-connection.js [hunk 1]
```diff
@@ -19,7 +20,20 @@ function ProxyConnection(browserConnection, janusConnection) {
   this.janusConnection = janusConnection;
 }
 
+/**
+ * @param {String} message
+ */
 ProxyConnection.prototype.processMessage = function(message) {
+  return this._processMessage(message).catch(function(error) {
```
Comment (tomaszdurka) [inline lib/proxy-connection.js]: Can we use then catch on new lines? Also for another one? Like this:
```js
this._processMessage(message)
  .catch(function(error) {
  })
  .then(function(error) {
  });
```

**Comment by tomaszdurka** [comment]
Rest looks good.

One thing to discuss - errors classes.
I feel strange that lower case `errors` stores multiple classes.
Maybe it should look rather like:
```js
var JanusError = require('janus-error');
throw new JanusError. Unauthorized(transaction);
```
@vogdb wdyt?

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@tomaszdurka @fvovan  how do you think should we close the proxy connection if an errors happens?

**Comment by vogdb** [comment]
paired with @tomaszdurka

**Pull request merged**